### PR TITLE
fix all linting errors and warnings

### LINKED ISSUE #185: Remove Linting Warnings
**Describe the bug**
We need to remove all the linting warnings that are coming

the thing to make is ```yarn lint``` will produce this and then we should fix this accordingly

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
Fixes #185  and all linting errors and lining warnings

image

fixed unused vars and added rue to check `no-explicit-any`, `react/destructuring-assignment`, `@typescript-eslint/explicit-module-boundary-types`

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
> Fixes #185 and all linting errors and lining warnings
> 
> image
> 
> fixed unused vars and added rue to check `no-explicit-any`, `react/destructuring-assignment`, `@typescript-eslint/explicit-module-boundary-types`

@yasharth291  do check this out

**Comment by Tejas-Ladhani** [comment]
Hey @yasharth291 @palisadoes, I think we should check the GitHub action's linting rule because everything is fine when running locally ( according to the linting rules defined in eslintrc.json.)

**Comment by Devesh21700Kumar** [comment]
> Hey @yasharth291 @palisadoes, I think we should check the GitHub action's linting rule because everything is fine when running locally ( according to the linting rules defined in eslintrc.json.)

@Tejas-Ladhani @yasharth291 
Yeah
This PR removes the warnings as well if any but the CI via actions is still showing an error.

**Comment by palisadoes** [comment]
@Devesh21700Kumar, @yasharth291 This used to work before. What changed? Is it the version of `ubuntu` or `yarn` in the GitHub Action? Is it something else? These things don't break by chance.

**Comment by yasharth291** [comment]
It is the problem with the linting workflow i think we should change it once @palisadoes

**Comment by palisadoes** [comment]
@yasharth291 So should we merge this PR after the conflicting files are fixed?

**Comment by yasharth291** [comment]
yes sir once he fetch the upstream

**Comment by palisadoes** [comment]
@Devesh21700Kumar There have been significant changes to the code base in recent days. Please merge your code with the current `upstream/develop` branch, fix the conflicting files (if any), and push the result to your repo. The PR will be automatically updated.

**Comment by Devesh21700Kumar** [comment]
> @Devesh21700Kumar There have been significant changes to the code base in recent days. Please merge your code with the current `upstream/develop` branch, fix the conflicting files (if any), and push the result to your repo. The PR will be automatically updated.

Yep on it
Thanks

**Review by Devesh21700Kumar — COMMENTED** [review]
@yasharth291 @palisadoes Done

**Comment by palisadoes** [comment]
@yasharth291 are the changes to `.eslintrc.json` acceptable? I'd prefer to make the linting more thorough, and I'm not sure this is the case here?

File: .eslintrc.json [hunk 1]
```diff
@@ -28,10 +28,23 @@
   // Specify the ESLint plugins tobe used
   "plugins": ["react", "@typescript-eslint", "react-hooks", "jest"],
   "rules": {
-    "react/destructuring-assignment": ["warn", "always"],
```
Comment (palisadoes) [inline .eslintrc.json:31]: @yasharth291 are the changes to `.eslintrc.json` acceptable? I'd prefer to make the linting more thorough, and I'm not sure this is the case here?

**Review by palisadoes — CHANGES_REQUESTED** [review]
@yasharth291 are the changes to `.eslintrc.json` acceptable? I'd prefer to make the linting more thorough, and I'm not sure this is the case here?

File: .eslintrc.json (hunk 1 above)
Comment (yasharth291) [reply .eslintrc.json:31]: I think this will not change anything as it is only warning

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
@yasharth291 Please approve and merge.

**Review by yasharth291 — APPROVED** [review]

**Pull request merged**